zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/PageTest.java

```diff
@@ -0,0 +1,145 @@
+package com.github.onsdigital.zebedee.api;
+
+import com.github.onsdigital.zebedee.Zebedee;
+import com.github.onsdigital.zebedee.content.page.MockPageUpdateHook;
+import com.github.onsdigital.zebedee.content.page.statistics.dataset.ApiDatasetLandingPage;
+import com.github.onsdigital.zebedee.content.util.ContentUtil;
+import com.github.onsdigital.zebedee.exceptions.NotFoundException;
+import com.github.onsdigital.zebedee.exceptions.ZebedeeException;
+import com.github.onsdigital.zebedee.model.Collections;
+import com.github.onsdigital.zebedee.reader.CollectionReader;
+import com.github.onsdigital.zebedee.util.ZebedeeCmsService;
+import org.apache.commons.fileupload.FileUploadException;
+import org.junit.Test;
+import org.springframework.mock.web.DelegatingServletInputStream;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PageTest extends ZebedeeAPIBaseTestCase {
```

Comment:
can you add test cases to cover session/collection not found exceptions & uri empty null?

Reply:
I have added the suggested test cases and more. Also refactored the tests to make them clearer 👍